src/helm/benchmark/metrics/metric.py

```diff
@@ -166,7 +149,7 @@ def evaluate(
 
             # Compute per-instance stats
             per_instance_stats: List[PerInstanceStats] = []
-            for instance, stats in zip(scenario_state.instances, results):
+            for instance, stats in zip(instances, results):
```

Comment:
I think switching this to `zip(request_state_sets, results)` would make it less fragile and more clear that we are putting the input and output of the parallel map back together.